config/initializers/event_bus.rb

```diff
@@ -63,23 +64,14 @@
     DiscussionReader.for_model(event.discussion, event.user).update_reader(
       volume: :loud,
       read_at: event.created_at
-    ) if event.discussion
-  end
-
-  config.listen('discussion_create') do |discussion|
-    DiscussionReader.for(user: discussion.author, discussion: discussion).update_reader(
-      volume: :loud,
-      read_at: discussion.created_at
     )
   end
 
-  config.listen('discussion_reader_viewed!',
-                'discussion_reader_dismissed!') do |discussion, actor|
-
-    reader_cache = Caches::DiscussionReader.new(user: actor, parents: Array(discussion))
-    collection = ActiveModel::ArraySerializer.new([discussion], each_serializer: MarkedAsRead::DiscussionSerializer, root: 'discussions', scope: { reader_cache: reader_cache } )
-
-    MessageChannelService.publish(collection, to: actor)
+  config.listen('discussion_reader_viewed!', 'discussion_reader_dismissed!') do |reader|
+    MessageChannelService.publish(
+      DiscussionReaderSerializer.new(reader, root: :discussions).as_json,
```

Comment:
I, for the life of me, could only vaguely follow why we did the removed lines 76-82 that way. Here we're saying, 'broadcast the newly read values to the person who read them', which makes your other tabs / devices auto-update when you read something.